src/ast/mod.rs

```diff
@@ -2013,9 +2013,19 @@ pub enum Statement {
     Truncate {
         #[cfg_attr(feature = "visitor", visit(with = "visit_relation"))]
         table_name: ObjectName,
+        table_names: Vec<ObjectName>,
```

Comment:
Design note: not totally happy with this, but `visit_relation` requires an `ObjectName`, and I didn't want to get caught attempting to refactor that code. `table_name` is populated during parse.

Reply:
hmm the workaround to have both `table_name` and `table_names` doesn't seem desirable. (extending `visit_relation` specifically I suspect might not be desirable either).
Could be that if a truncate can reference multiple tables then maybe its no longer eligible for the visitor functionality and we can remove the derive instead (replacing `table_name` with `table_names`) - just thinking out loud based on how similar statements like `UPDATE` seem to be treated in this regard today which seemed to suggest there's no support currently for such cases? But yeah, I'm not familiar with the use cases to tell what makes sense, cc @alamb @lovasoa if you have thoughts?

Reply:
Could we work around the need to call `visit_relation` on a single object by adding another layer of indirection? Something like

```rust
/// Target of a `TRUNCATE TABLE` command
/// 
/// Note this is its own struct because `visit_relation` requires an `ObjectName` (not a `Vec<ObjectName>`)
struct TruncateTableTarget {
  /// name of the table being truncated
  #[cfg_attr(feature = "visitor", visit(with = "visit_relation"))]
  name: ObjectName,
}

pub enum Statement { 
  ...
  Truncate {
    table_names: Vec<TruncateTableTarget>,
    ....
  }
}
```
```